synapseclient/credentials/cached_sessions.py

```diff
@@ -0,0 +1,88 @@
+import keyring
+import os
+import json
+import warnings
+from keyring.errors import PasswordDeleteError
+from keyring.backends.fail import Keyring as FailKeyring
+
+
+SYNAPSE_CACHED_SESSION_APLICATION_NAME = "<SYNAPSE.ORG CLIENT>"
+SESSION_CAHCE_FILEPATH = os.path.expanduser("~/.synapseSession")
+
+# In the case where no key store backend is available (most likely in Linux), a fail.Keyring is returned which will throw errors when any of its functions are called
+# However, the errors thrown are of the generic type RuntimeError (not a subclass of RuntimeError).
+# It didn't feel safe to try/except and ignore RuntimeError since there are very many other RuntimeErrors that could occur.
+# Instead, we use this boolean to keep track of whether we should call the keyring methods
+_keyring_is_available = not isinstance(keyring.get_keyring(), FailKeyring)
+
+
+def get_api_key(username):
+    """
+    Retrieves the user's API key
+    :param str username:
+    :return: API key for the specified username
+    :rtype: str
+    """
+    if _keyring_is_available:
+        return keyring.get_password(SYNAPSE_CACHED_SESSION_APLICATION_NAME, username)
+    return None
+
+
+def remove_api_key(username):
+    if _keyring_is_available:
+        try:
+            keyring.delete_password(SYNAPSE_CACHED_SESSION_APLICATION_NAME, username)
+        except PasswordDeleteError:
+            #The API key does not exist, but that is fine
+            pass
+
+
+def set_api_key(username, api_key):
+    if _keyring_is_available:
+        keyring.set_password(SYNAPSE_CACHED_SESSION_APLICATION_NAME, username, api_key)
+    else:
+        warnings.warn("\nUnable to save user credentials as you do not have a keyring available.\n"
+                      "'GNOME keyring' is necessary to be installed for credential store to work on Linux systems:\n"
```

Comment:
Move this content to doc (under login() > rememberMe param). This warning should have a link to the doc.